### PR TITLE
O-RAN O2 Application in StarlingX

### PR DESCRIPTION
**Pull request by gaolitao**
Add a blogpost for O2 interface feature in StarlingX 8.0

### CODE REVIEW

File: src/pages/blog/starlingx-oran-o2-application.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+O-Cloud can consist of multiple Deployment Management Services (DMS). Each DMS can manage leased resources from multiple resource pools and span multiple locations. There is a single Infrastructure Management Services (IMS) for O-Cloud that manages all resources of DMSes as well as resources that are not allocated to any DMS in the O-Cloud.
+
+SMO can utilize blueprints or other pre-configuration templates to assign resources to resource pools and to assign resource to a DMS. Once the O-Cloud is operational, following functions can be performed over the O2 interfaces as:
+
+- O-Cloud Infrastructure Resource Management
+  - O-Cloud infrastructure Discovery and administration
+  - O-Cloud infrastructure Scale-In, Scale-Out
+  - O-Cloud infrastructure FCAPS (PM, CM, FM, Communication Surveillance)
+  - O-Cloud infrastructure Platform Software Management
+- Managing Abstracted Resources and DMSes
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-o2-application.md:41]:
```suggestion
- Managing Abstract Resources and DMSs
```

File: src/pages/blog/starlingx-oran-o2-application.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+  - Creation, Scale-In, Scale-Out of abstracted assigned O-Cloud infrastructure resources
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-o2-application.md:42]:
```suggestion
  - Creation, Scale-In, Scale-Out of assigned O-Cloud infrastructure resources
```

File: src/pages/blog/starlingx-oran-o2-application.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+  - Deployment FCAPS (PM, FM) for abstracted O-Cloud infrastructure resources
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-o2-application.md:43]:
```suggestion
  - Deployment of FCAPS components for O-Cloud infrastructure resources
```

File: src/pages/blog/starlingx-oran-o2-application.md [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  - Deployment DMS (Creation, Deletion and Lease of O-Cloud infrastructure)
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-o2-application.md:44]:
```suggestion
  - Deployment of DMS (Creation, Deletion and Lease of O-Cloud infrastructure)
```

File: src/pages/blog/starlingx-oran-o2-application.md [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+- NF & Services Deployment Orchestration
+  - Deployment Software Management
+  - Deployment, Termination, Scaling, and Healing of NF & Services deployment resources
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-o2-application.md:47]:
```suggestion
  - Deployment, Termination, Scaling, and Healing of NF & Services resources
```

File: src/pages/blog/starlingx-oran-o2-application.md [hunk 6]
```diff
@@ -0,0 +1,82 @@
...
+  - FCAPS (PM, FM) for NF & Services deployment resources
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-o2-application.md:48]:
```suggestion
  - FCAPS, for instance PM and FM, for NF & Services deployment resources
```

File: src/pages/blog/starlingx-oran-o2-application.md [hunk 7]
```diff
@@ -0,0 +1,82 @@
...
+
+With this new feature, telecom providers can leverage StarlingX's powerful cloud infrastructure capabilities while maintaining the flexibility to choose their preferred network functions from different vendors. This allows them to easily customize their 5G network to meet their specific needs and requirements, without being locked into a single vendor's solution.
+
+StarlingX's O-RAN specification compliant O2 interfaces feature is built on top of the[ O2 Service in G release](https://docs.o-ran-sc.org/projects/o-ran-sc-pti-o2/en/g-release/overview.html) of Open RAN Software Community (ORAN-SC) INF project, where *O-RAN O2ims Interface Specification 3.0* and *O-RAN O2dms Interface Specification: Kubernetes Native API Profile for Containerized NFs 2.0* are the specifications which are compliant to, and these are publised in October 2022 and are the latest version of the specification when this blogpost is being written.
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-o2-application.md:56]:
```suggestion
StarlingX's O-RAN specification compliant O2 interfaces feature is built on top of the [O2 Service in G release](https://docs.o-ran-sc.org/projects/o-ran-sc-pti-o2/en/g-release/overview.html) of Open RAN Software Community (ORAN-SC) INF project. At the time of publishing this blog post, the latest version of the compliant specifications are *O-RAN O2ims Interface Specification 3.0* and *O-RAN O2dms Interface Specification: Kubernetes Native API Profile for Containerized NFs 2.0*.
```

File: src/pages/blog/starlingx-oran-o2-application.md [hunk 8]
```diff
@@ -0,0 +1,82 @@
...
+
+The O-RAN O2 service includes a set of software components that enable the functionality of O-RAN compliant O2 interfaces:
+
+* O2API: This component provides O2 IMS interfaces servicing external requests.
+* Database: This component stores the O2 infrastructure inventory Information.
+* MessageQueue: This component is used to exchange asynchronous messages.
+* Watcher: This component provides the capability of O-Cloud resource audit.
+* PubSub: This component handles notification related logics, like subscribe and notify.
+
+The O-RAN O2 service is integrated into StarlingX as a system application. With the [StarlingX Application Package Manager](https://docs.starlingx.io/system_configuration/kubernetes/system-configuration-starlingx-application-package-manager.html), O-RAN O2 service application's lifecycle can be managed easily, including managing overrides to the helm charts within the application.
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-o2-application.md:66]:
```suggestion
The O-RAN O2 service is integrated into StarlingX as a system application. With the [StarlingX Application Package Manager](https://docs.starlingx.io/system_configuration/kubernetes/system-configuration-starlingx-application-package-manager.html), the O-RAN O2 service application's lifecycle can be managed easily, including managing overrides to the helm charts within the application.
```

File: src/pages/blog/starlingx-oran-o2-application.md [hunk 9]
```diff
@@ -0,0 +1,82 @@
...
+
+A StarlingX application package is a compressed tarball containing a metadata.yaml file, a manifest.yaml FluxCD manifest file, and a charts directory containing helm charts and a checksum.md5 file. The metadata.yaml file contains the application name, version, and optional helm repository and disabled charts information.
+
+The O-RAN O2 application package is saved in StarlingX during system installation, but it is not applied by default. When system administrators plan to integrate StarlingX with a SMO application that performs management of O-Cloud infrastructure and the deployment lifecycle management of O-RAN cloudified NFs, they can just apply and put O-RAN O2 application into service by using a small number of `system` CLI commands.
+
+Please refer to the detailed [procedure for how to install and uninstall O-RAN O2 application in StarlingX](https://docs.starlingx.io/admintasks/kubernetes/oran-o2-application-b50a0c899e66.html).
```
Comment (ildikov) [inline src/pages/blog/starlingx-oran-o2-application.md:72]:
```suggestion
Please refer to the detailed [procedure for how to install and uninstall the O-RAN O2 application in StarlingX](https://docs.starlingx.io/admintasks/kubernetes/oran-o2-application-b50a0c899e66.html).
```

**Review by ildikov — CHANGES_REQUESTED** [review]
Hi,

Thank you for submitting this blog post to describe the new O-RAN O2 Application in StarlingX. I added a few suggestions in line to correct typos and further polish the text.

Please check my comments and let me know if you have any questions to them.

I'm hoping to publish the blog early next week, once the comments are resolved.

Thanks and Best Regards,
Ildikó

**Review by ildikov — APPROVED** [review]
This looks great, thank you for updating the post.

I will publish the blog post on Monday in US morning hours.

Thanks and Best Regards,
Ildikó

**Comment by gaolitao** [comment]
@ildikov  Thank you for merging, seems that the website cannot render the 3rd level of list in https://www.starlingx.io/blog/starlingx-oran-o2-application/

image

**Comment by ildikov** [comment]
@gaolitao Thank you for catching that! I just reached out to our design team to check what the issue might be with the formatting. They're looking into the fix, I'm hoping that it will be corrected early next week.

**Pull request merged**